src/sources.cu

```diff
@@ -18,12 +19,28 @@ double OmK2(GridRef& g, double Mstar, int i, int j) {
     return GMsun * Mstar / std::pow(g.Rc(i)*g.Rc(i)+g.Zc(i,j)*g.Zc(i,j), 1.5);
 
 }
+
+template<bool full_stokes>
 __device__
-double t_s(double rho_g, double rho_m, double s, double T, double mu) {
+double t_s(Prims wd, Prims wg, double rho_m, double s, double T, double mu) {
 
-    double v_th = sqrt(8*k_B*T / (mu*m_H*M_PI));
+    double mfp = mu * m_H / (wg.rho * 2.e-15);
+    double cs = sqrt(k_B*T / (mu*m_H));
 
-    return rho_m * s / (rho_g * v_th);
+    if (full_stokes) {
+        double u_rel = sqrt((wd.v_phi-wg.v_phi)*(wd.v_phi-wg.v_phi) 
+                                + (wd.v_R-wg.v_R)*(wd.v_R-wg.v_R) + (wd.v_Z-wg.v_Z)*(wd.v_Z-wg.v_Z));
+        return 2.66666666667 * rho_m * s / (wg.rho * calc_C_D(s,wg.rho,cs,u_rel,mu) * u_rel);
+    }
+    else {
+        if (s<2.25*mfp) 
```

Comment:
I cover the epstein regime before using the drag constant calculation, to avoid having to calculate the relative velocity if its isnt needed.

Reply:
I was initially using the drag constant calculation to cover all regimes, which is why it's still in the function - I'd forgotten to remove it.

Reply:
OK - but this approach is prone to error because the user must check the regimes themselves. I suggest providing a single function that computes t_stop for all regimes. It can decide internally whether the velocity is needed and compute |v| if so. If you want two versions (a simple and more accurate one) that's fine, but the simple one is probably ok anyway.

Reply:
I'm not sure that I understand what you mean - the user just decides whether to use the slower, more accurate analytic form or the simpler version with epstein and three stokes regimes. But you're right that I can neaten it up by putting all of the logic in one function, so I'll do that.

Reply:
The issue is that if anyone wants to compute t_s for some other reason, they need to check the mean-free path by hand and decide on which region to use, which is error-prone. Is it possible to make t_s from sources.cu the available to everyone (in drag_const.h)? I would then call that t_s function everywhere so that the user does not need to check.

Reply:
Right, yes I see now. What you've suggested is what I've been changing it to, will push changes after I've finished testing.